### DISCUSSION TITLE
USB boot

### DISCUSSION THREAD

**Discussion by craigyjp** [rsta2/circle#341] [General]
Can circle be booted from USB? Just wondering as it would make a build that I have so much easier to have the OS and app on a USB stick rather than trying to get access to the SD for updates.

**Comment by rsta2** [comment]
Yes. The Raspberry Pi can be booted from USB. Please see the instructions [here](https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#booting-from-usb-mass-storage), which depend on the RPi model used. This is not different for Circle as for other OSes. For most Circle samples, which do not access the SD card, you don't need to do a special build. For the other samples, you may have to redirect the SD card access to USB. For [this sample](https://github.com/rsta2/circle/tree/master/addon/fatfs/sample) this is described in the README.

**Comment by mgnthe54** [comment]
Hello, I'm not really sure I'm in the right place, but this is the discussion closest to my concern!
I have a raspberry pi 4 running bullseye 64 bit.
I configured the boot on USB (SSD disk with external power supply to be sure that it is operational before restarting the PI) then if not available on SD:
sudi raspi-config ->option 6 Advanced Options
option A6 Boot order
B2 USB boot.
In the micro SD slot of the raspberry I have an SD containing a clone (via SD Card Copier) of my SSD disk.
If I reboot my Raspberry I start each time on the SD (unless the SD is not in the slot of course!)
How to boot from the SSD via USB when a micro SD is in the Pi?

THANKS

**Reply by ArnoldUK** [reply]
There is a special bootcode.bin that can be placed onto the micro-sd card which allows booting from USB drives but I'm not sure if it would in your case. You can always try it anyway.
[special boot modes](https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-boot-modes)
[bootcode.bin](https://github.com/raspberrypi/firmware/raw/master/boot/bootcode.bin)

**Reply by rsta2** [reply]
This discussion has a duplicate at issue #365. It will be continued there.
